### PR TITLE
more improvements to the review screen!~

### PR DESCRIPTION
**Pull request by wendybujalski**
## How does this PR change the system?

- Basic nesting of AV diffs on the review screen using a tree
- Better display of the AV diff data, subscriptions use the same styling as the AttributePanel
- Revert button works and doesn't show in most cases where it can't work, still needs improvements
- Overall better UI styling

Also some very minor unrelated UI fixes.

#### Screenshots:

Screenshot 2025-08-22 at 1 04 09 PM

#### Out of Scope:

Still more work to do, see ENG-3197 for the details on what is left

## How was it tested?

Lots of poking around with different components, will definitely need more hands to test this and find edge cases. But it is still behind a feature flag and there should not be any major breaking bugs, just UI jank.

### CODE REVIEW

File: app/web/src/components/CustomizeTabs.vue [hunk 1]
```diff
@@ -8,8 +8,10 @@
   >
     <TabGroupItem slug="assets">
       <template #label>
-        Assets Installed
-        <PillCounter :count="assetList.length" />
+        <div class="flex flex-row items-center gap-xs">
+          <div>Assets Installed</div>
+          <PillCounter :count="assetList.length" />
+        </div>
```
Comment (nickgerace) [inline app/web/src/components/CustomizeTabs.vue:14]: This is only used in module land right? We can keep it, just curious for what this impacts.

File: app/web/src/newhotness/Review.vue [hunk 2]
```diff
@@ -283,6 +309,37 @@ const componentDiffQueries = useQueries({
...
+  const result = changeSetComponentList.value.map((component) => {
+    // TEMPORARY: "fix" issues in the MV so we can do better testing / display in the short term.
+    // Ultimately, we will fix the MV instead.
+    // const diff = fixComponentDiff(componentDiffs[component.id]);
+    return {
+      ...component,
+      // diff,
+      // diffStatus: diff?.diffStatus ?? component.diffStatus,
+    };
+  });
```
Comment (nickgerace) [inline app/web/src/newhotness/Review.vue:327]: Why not fix the MV? Just fiddly?

File: app/web/src/newhotness/Review.vue [hunk 3]
```diff
@@ -301,28 +358,133 @@ const componentList = computed(() => {
...
-  // Add any head components that aren't in the list as "removed"
-  for (const headComponent of headComponentList.value) {
-    if (!result.find((c) => c.id === headComponent.id)) {
-      result.push({
-        ...headComponent,
-        diff: undefined,
-        diffStatus: "Removed" as const,
-      });
+/** A tree version of AttributeDiff:
+ * {
```
Comment (nickgerace) [inline app/web/src/newhotness/Review.vue:376]: Nice doc comment!

File: app/web/src/newhotness/Review.vue [hunk 4]
```diff
@@ -301,28 +358,133 @@ const componentList = computed(() => {
...
+  return true;
+}
+
+function fixAttributeSourceAndValue(sourceAndValue?: AttributeSourceAndValue) {
+  if (!sourceAndValue) return undefined;
+  const { $source } = sourceAndValue;
+  if ($source.prototype !== undefined) {
+    // subscription to /domain/region on Region changed (01K2N2CFSAE0PPHNJ31SKP97HH)
+    const { prototype, ...otherFields } = $source;
+    const match = prototype.match(/^subscription to (\/.+) on .+ \((\w+)\)/);
```
Comment (nickgerace) [inline app/web/src/newhotness/Review.vue:467]: oh my

File: app/web/src/newhotness/ReviewAttributeItemSourceAndValue.vue [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+      "
+    >
+      {{ old ? "-" : "+" }}
+    </div>
+    <!-- <div class="text-2xs">
+      {{ rawValue }} / {{ rawSource }}
+    </div> -->
+    <TruncateWithTooltip
+      v-if="displayKind === 'value' || displayKind === 'complex'"
+      :class="clsx('py-2xs', old && 'line-through')"
```
Comment (nickgerace) [inline app/web/src/newhotness/ReviewAttributeItemSourceAndValue.vue:28]: Neat! Good way to do the strikethrough. I'll probably need it for component history

File: app/web/src/newhotness/ReviewAttributeItemSourceAndValue.vue [hunk 6]
```diff
@@ -0,0 +1,102 @@
...
+}>();
+
+const rawValue = computed(() => props.sourceAndValue.$value);
+const rawSource = computed(() => props.sourceAndValue.$source);
+
+const displayKind = computed<DisplayKind>(() => {
+  if ("component" in rawSource.value) return "subscription";
+  else if ("value" in rawSource.value) return "value";
+  // TODO(Wendy) - complex AV diffs?
+  return "hidden";
```
Comment (nickgerace) [inline app/web/src/newhotness/ReviewAttributeItemSourceAndValue.vue:96]: Does this block maps / arrays / objects? I'm a bit confused to what this does.

File: app/web/src/newhotness/Review.vue [hunk 7]
```diff
@@ -126,19 +126,43 @@
       </div>
     </div>
     <div class="main flex flex-col gap-xs m-xs">
-      <CollapsingFlexItem v-if="selectedComponent" disableCollapse>
+      <CollapsingFlexItem
+        v-if="selectedComponentId && selectedComponent"
```
Comment (jobelenus) [inline app/web/src/newhotness/Review.vue:130]: feels a tad redundant?

File: app/web/src/components/CustomizeTabs.vue (hunk 1 above)
Comment (wendybujalski) [reply app/web/src/components/CustomizeTabs.vue:14]: Ah yeah, I just fixed the alignment of the `PillCounter` because it was bugging me 😅

File: app/web/src/newhotness/Review.vue (hunk 2 above)
Comment (wendybujalski) [reply app/web/src/newhotness/Review.vue:327]: Yeah this is much faster than doing the fix on the MV itself

File: app/web/src/newhotness/Review.vue [hunk 8]
```diff
@@ -126,19 +126,43 @@
...
+            "
+          >
+            <ReviewAttributeItem
+              v-for="(item, name) in selectedComponent.attributeDiffTree
+                .children.domain.children"
+              :key="name"
+              :selectedComponentId="selectedComponentId"
+              :name="name"
+              :item="item"
+            />
```
Comment (jobelenus) [inline app/web/src/newhotness/Review.vue:163]: I assume the `/secrets/` tree is still to do?

File: app/web/src/newhotness/ReviewAttributeItemSourceAndValue.vue (hunk 6 above)
Comment (wendybujalski) [reply app/web/src/newhotness/ReviewAttributeItemSourceAndValue.vue:96]: No, this just filters out any `SourceAndValue` info that isn't a raw value or a subscription. Arrays will display properly with nesting children, though I still need to filter out their data when it is redundant with the displayed children.

File: app/web/src/newhotness/Review.vue (hunk 7 above)
Comment (wendybujalski) [reply app/web/src/newhotness/Review.vue:130]: Prevents typescript complaints, could probably fix another way 😅

File: app/web/src/newhotness/Review.vue (hunk 8 above)
Comment (wendybujalski) [reply app/web/src/newhotness/Review.vue:163]: Yep!

File: app/web/src/newhotness/Review.vue [hunk 9]
```diff
@@ -283,6 +309,37 @@ const componentDiffQueries = useQueries({
...
+    // const diff = fixComponentDiff(componentDiffs[component.id]);
+    return {
+      ...component,
+      // diff,
+      // diffStatus: diff?.diffStatus ?? component.diffStatus,
+    };
+  });
+
+  // Add any head components that aren't in the list as "removed"
+  for (const headComponent of headComponentList.value) {
```
Comment (jobelenus) [inline app/web/src/newhotness/Review.vue:330]: i thought the MV gave us this data?

Comment (wendybujalski) [reply app/web/src/newhotness/Review.vue:330]: We can't get data on components which do not exist in the change set from the MV so we have to grab that data from Head separately.

Comment (britmyerss) [reply app/web/src/newhotness/Review.vue:330]: I think I've got this covered in the MV i am building - we'll have a list of ComponentDiffs for Components that only exist on head

File: app/web/src/newhotness/Review.vue (hunk 4 above)
Comment (jobelenus) [reply app/web/src/newhotness/Review.vue:467]: how come we have to do this? why doesn't the MV give us this as more structured data?

**Review by jobelenus — APPROVED** [review]
